src/validators/enum_.rs

```diff
@@ -183,8 +183,14 @@ impl EnumValidateValue for PlainEnumValidator {
                         } else if py_input.is_instance_of::<PyFloat>() {
                             return Ok(lookup.validate_int(py, input, false)?.map(|v| v.clone_ref(py)));
                         }
+                        if py_input.is_instance_of::<PyAny>() {
+                            if let Ok(Some(res)) = lookup.try_validate_any(input) {
+                                return Ok(Some(res.clone_ref(py)));
+                            }
+                        }
```

Comment:
I think instead of adding this new `try_validate_any` function which attempts to reproduce some Python logic in Rust, what we should do instead is try to let the Python enum class create the value. That way any functions like `__missing__` in the enum validation also get a chance to participate.

So in Python we'd be calling `MyEnum(py_input)`, and probably discarding exceptions.

The equivalent Rust expression here would be something like `if let Ok(res) = class.call1((py_input,)) { return Ok(res) }`. Where `class` is currently stored on the `EnumValidator`. You'll need to modify the `EnumValidateValue` trait to pass `class` in.

Reply:
Yeah 💯, it makes sense and is simple.